src/System.Net.Sockets/src/System/Net/Sockets/Socket.Tasks.cs

```diff
@@ -140,25 +140,36 @@ internal Task<int> ReceiveAsync(ArraySegment<byte> buffer, SocketFlags socketFla
             if (saea.BufferList != null) saea.BufferList = null;
             saea.SetBuffer(buffer.Array, buffer.Offset, buffer.Count);
             saea.SocketFlags = socketFlags;
-            saea.WrapExceptionsInIOExceptions = wrapExceptionsInIOExceptions;
+            saea._wrapExceptionsInIOExceptions = wrapExceptionsInIOExceptions;
 
             // Initiate the receive
             Task<int> t;
+            bool responsibleForReturningToPool;
```

Comment:
Re this pattern, I think it's slightly clearer to encapsulate the logic for sync vs async completion, e.g.:

```
if (!ReceiveAsync(saea))
{
    t = GetSyncCompletionTask(saea);
}
else
{
    t = GetPendingTask(saea);
}
```

Or even just

```
t = GetTask(saea, ReceiveAsync(saea));
```

Thoughts?